core/templates/pages/collection-player-page/collection-footer/collection-footer.component.spec.ts

```diff
@@ -0,0 +1,81 @@
+// Copyright 2021 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Unit tests for collection footer component.
+ */
+
+import { NO_ERRORS_SCHEMA } from '@angular/core';
+import { TestBed, waitForAsync, ComponentFixture } from '@angular/core/testing';
+import { MockTranslatePipe } from 'tests/unit-test-utils';
+import { UrlInterpolationService } from
+  'domain/utilities/url-interpolation.service';
+import { UrlService } from 'services/contextual/url.service';
+import { CollectionFooterComponent } from './collection-footer.component';
+
+describe('Collection footer component', () => {
+  let urlInterpolationService: UrlInterpolationService;
+  let urlService: UrlService;
+  let component: CollectionFooterComponent;
+  let fixture: ComponentFixture<CollectionFooterComponent>;
+
+  beforeEach(waitForAsync(() => {
+    TestBed.configureTestingModule({
+      declarations: [
+        CollectionFooterComponent,
+        MockTranslatePipe
+      ],
+      providers: [
+        UrlInterpolationService,
+        UrlService
+      ],
+      schemas: [
+        NO_ERRORS_SCHEMA
+      ]
+    }).compileComponents();
+  }));
+
+  beforeEach(() => {
+    fixture = TestBed.createComponent(CollectionFooterComponent);
+    component = fixture.componentInstance;
+    urlInterpolationService = TestBed.inject(UrlInterpolationService);
+    urlInterpolationService = (urlInterpolationService as unknown) as
+      jasmine.SpyObj<UrlInterpolationService>;
+    urlService = TestBed.inject(UrlService);
+    urlService = (urlService as unknown) as jasmine.SpyObj<UrlService>;
+  });
+
+  it('should load the component properly on playing a collection', () => {
+    spyOn(urlService, 'getCollectionIdFromUrl').and.returnValue('abcdef');
+    expect(component.collectionId).toBe('');
+
+    component.ngOnInit();
+    expect(urlService.getCollectionIdFromUrl).toHaveBeenCalled();
+    expect(component.collectionId).toBe('abcdef');
+  });
+
+  it('should get the static image url from the image path', () => {
+    spyOn(urlService, 'getCollectionIdFromUrl').and.returnValue('abcdef');
+    spyOn(urlInterpolationService, 'getStaticImageUrl');
+
+    component.ngOnInit();
+    fixture.whenStable()
```

Comment:
Place a comment explaining why `fixture.whenStable()` is required here

Reply:
I added a return value to `urlInterpolationService` spy and now removing `fixture.whenStable()` works. Removed it now